evil-common.el

```diff
@@ -2094,15 +2096,16 @@ register instead of replacing its content."
    ((member register '(?: ?. ?%))
     (user-error "Can't modify read-only register"))
    ((eq register ?\")
-    (kill-new text))
+    (kill-new text)
+    (evil-set-register ?0 text))
```

Comment:
I think it makes more sense to check for `register` equals `""` in `evil-kill-new` instead.